functions/notify_slack.py

```diff
@@ -583,7 +583,7 @@ def parse_notification(message: Dict[str, Any], subject: Optional[str], region:
     :params region: AWS region where the event originated from
     :returns: Slack message payload
     """
-    if "AlarmName" in message:
+    if isinstance(message, Dict) and "AlarmName" in message:
         return format_cloudwatch_alarm(message=message, region=region)
```

Comment:
`format_cloudwatch_alarm()` is assuming it is receiving a `Dict`, so going the `isinstance(message, Dict)` check here is correct.